android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -21,15 +21,164 @@ import com.uber.rib.core.lifecycle.InteractorEvent
 import com.uber.rib.core.lifecycle.PresenterEvent
 import com.uber.rib.core.lifecycle.WorkerEvent
 import io.reactivex.Observable
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.channels.BufferOverflow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.merge
 import kotlinx.coroutines.flow.transformWhile
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.rx2.asFlow
 import kotlinx.coroutines.rx2.asObservable
+import kotlin.system.measureTimeMillis
 
 /** Helper class to bind to an interactor's lifecycle to translate it to a [Worker] lifecycle. */
 object WorkerBinder {
+
+  @JvmStatic
+  private var ribWorkerBindMonitoringListeners: MutableSet<RibWorkerBindMonitoringListener> = mutableSetOf()
+
+  /**
+   * Adds [RibWorkerBindMonitoringListener]
+   */
+  @JvmStatic
+  fun addMonitoringListener(ribWorkerBindMonitoringListener: RibWorkerBindMonitoringListener) {
+    this.ribWorkerBindMonitoringListeners.add(ribWorkerBindMonitoringListener)
+  }
+
+  /**
+   * Bind a worker (ie. a manager or any other class that needs an interactor's lifecycle) to an
+   * interactor's lifecycle events.
+   *
+   * Worker.onStart/onStop will run on the specified [RibWorkerThreadingType]
+   *
+   * @param worker The class that wants to be informed when to start and stop doing work.
+   * @param workerLifecycle Observable<WorkerEvent>
+   * @param ribWorkerThreadingType Rib Threading type where Worker.onStart/onStop will operate
+   */
+  @JvmStatic
+  fun bind(
+    workerToBeBound: Worker,
```

Comment:
I'd just call the parameter `worker` instead, it's obvious it's the worker to be bound.

Reply:
Totally agree, updated as well